### ISSUE TITLE
Improper Initialization of Loss Metrics Derived from `MultiHorizonMetric` when Moving from GPU to CPU

### ISSUE THREAD

**Issue by B-Deforce** [jdb78/pytorch-forecasting#1355]
- PyTorch-Forecasting version: 1.0.0
- PyTorch version: 2.0.1+cpu
- torchmetrics: 0.11.4
- Python version: 3.11.3
- Operating System: Windows 11

### Expected behavior

I trained a model on GPU that included loss metrics (`MAE`, `MASE`, etc.) inheriting from `MultiHorizonMetric`, saved the model to a PyTorch Lightning checkpoint, and then attempted to load the model from the checkpoint on a CPU. I expected the model to load without any issues.

### Actual behavior

However, an `AttributeError` occurred due to a missing `_dtype_convert` attribute when the model was moved from GPU to CPU. I believe this is related to the improper initialization of the loss classes, as the `_apply` method in the loading process expects this attribute.

### Code to reproduce the problem

```python
# Initialization and saving process
model = MyModel()  # model that includes MAE or MASE
trainer = Trainer(gpus=1)  # ensure model is trained on GPU
trainer.fit(model, dataloader)

# Loading process
model = MyModel.load_from_checkpoint(checkpoint_path="path_to_checkpoint.ckpt", map_location=torch.device('cpu'))
```
I found that adding `super().__init__(**kwargs)` at the beginning of the `__init__` method for these classes (e.g. for `MAE`) resolved the issue, suggesting that some necessary initialization from the parent `MultiHorizonMetric` or grandparent `Metric` classes was skipped. I think this can be resolved fairly easy by adding `super().__init__(**kwargs)` to the losses where it's not present (e.g. `MAE`, `MASE`, ...)
